### PR TITLE
feat: add exception for exposure bigger than fps

### PR DESCRIPTION
**Pull request by eduardacoppo**
On-top of https://github.com/tidewise/drivers-orogen-camera_lucid/pull/17

# What is this PR for

[sc-63296]

# How I did it

# Results, How I tested

https://github.com/user-attachments/assets/c0f8fea1-0a64-4a3e-a266-990b3f79baa9

# Checklist
- [x] Assign yourself  in the PR

### CODE REVIEW

**Review by higor-melo — APPROVED** [review]

**Review by higor-melo — COMMENTED** [review]
Can you create a test for it? (Had to re-request review since i mistakenly approved it)

File: tasks/Task.cpp [hunk 1]
```diff
@@ -845,6 +845,12 @@ void Task::exposureConfiguration(Arena::IDevice& device)
         }
     }
     else if (image_config.exposure_auto == ExposureAuto::EXPOSURE_AUTO_OFF) {
+        double exposure_time_seconds = image_config.exposure_time.toSeconds();
+        double seconds_per_frame = 1 / image_config.frame_rate;
+        if (exposure_time_seconds > seconds_per_frame + 0.0001) {
+            exception(EXPOSURE_TIME_SURPASSES_FPS);
+            return;
+        }
```
Comment (doudou) [inline tasks/Task.cpp]: Same comment than with the other pull request ... Throw an exception.

File: tasks/Task.cpp [hunk 2]
```diff
@@ -845,6 +845,12 @@ void Task::exposureConfiguration(Arena::IDevice& device)
         }
     }
     else if (image_config.exposure_auto == ExposureAuto::EXPOSURE_AUTO_OFF) {
+        double exposure_time_seconds = image_config.exposure_time.toSeconds();
+        double seconds_per_frame = 1 / image_config.frame_rate;
+        if (exposure_time_seconds > seconds_per_frame + 0.0001) {
```
Comment (doudou) [inline tasks/Task.cpp]: Actually ... if anything, the tolerance should be going the opposite way. I would guess that one needs the exposure to be slightly smaller than the seconds per frame, not longer.

File: tasks/Task.cpp [hunk 3]
```diff
@@ -845,6 +845,12 @@ void Task::exposureConfiguration(Arena::IDevice& device)
         }
     }
     else if (image_config.exposure_auto == ExposureAuto::EXPOSURE_AUTO_OFF) {
+        double exposure_time_seconds = image_config.exposure_time.toSeconds();
+        double seconds_per_frame = 1 / image_config.frame_rate;
+        if (exposure_time_seconds > seconds_per_frame - 0.0001) {
+            throw std::runtime_error("EXPOSURE TIME SURPASSES FPS");
```
Comment (doudou) [inline tasks/Task.cpp]: While the all caps

"surpasses" -> "is bigger than the acquisition period needed by the configured frame_rate"

File: tasks/Task.cpp [hunk 4]
```diff
@@ -845,6 +845,12 @@ void Task::exposureConfiguration(Arena::IDevice& device)
...
+            return;
```
Comment (doudou) [inline tasks/Task.cpp]: Same comment about throw terminating the control flow.

**Comment by eduardacoppo** [comment]
> Can you create a test for it? (Had to re-request review since i mistakenly approved it)

Added a video with the test in the description

File: tasks/Task.cpp [hunk 5]
```diff
@@ -845,6 +845,12 @@ void Task::exposureConfiguration(Arena::IDevice& device)
         }
     }
     else if (image_config.exposure_auto == ExposureAuto::EXPOSURE_AUTO_OFF) {
+        double exposure_time_seconds = image_config.exposure_time.toSeconds();
+        double seconds_per_frame = 1 / image_config.frame_rate;
+        if (exposure_time_seconds > seconds_per_frame - 0.0001) {
+            throw std::runtime_error("EXPOSURE TIME IS BIGGER THAN THE ACQUISITION "
+                                     "PERIOD NEEDED BY THE CONFIGURED FRAME_RATE");
```
Comment (doudou) [inline tasks/Task.cpp:852]: No need to shout ! :P

**Review by doudou — APPROVED** [review]

**Pull request merged**